jose/src/main/kotlin/web5/sdk/jose/jwt/Jwt.kt

```diff
@@ -0,0 +1,240 @@
+    } catch (e: Exception) {
+      throw SignatureException(
+        "Malformed JWT. " +
+          "Invalid base64url encoding for JWT payload. ${e.message}"
+      )
+    }
+
+    return DecodedJwt(
+      header = decodedJws.header,
+      claims = claims,
+      signature = decodedJws.signature,
+      parts = decodedJws.parts
+    )
+
+  }
+
+  /**
+   * Sign a JwtClaimsSet using a Bearer DID.
+   *
+   * @param did The Bearer DID to sign with
+   * @param payload The JwtClaimsSet payload to sign
+   * @return The signed JWT
+   */
+  public fun sign(did: BearerDid, payload: JwtClaimsSet): String {
+    val jwtModule = SimpleModule()
+      .addSerializer(
+        JwtClaimsSet::class.java,
+        JwtClaimsSetSerializer()
+      )
+    Json.jsonMapper.registerModule(jwtModule)
+    val payloadJsonString = Json.jsonMapper.writeValueAsString(payload)
+    val payloadBytes = Convert(payloadJsonString).toByteArray()
+
+    return Jws.sign(did, payloadBytes)
+  }
+
+  /**
+   * Verify a JWT.
+   *
+   * @param jwt The JWT to verify
+   * @return DecodedJwt
+   */
+  public fun verify(jwt: String): DecodedJwt {
+    val decodedJwt = decode(jwt)
+    decodedJwt.verify()
+    return decodedJwt
+  }
+}
+
+/**
+ * DecodedJwt is a compact JWT decoded into its parts.
+ *
+ * @property header The JWT header
+ * @property claims The JWT claims
+ * @property signature The JWT signature
+ * @property parts The JWT parts
+ */
+public class DecodedJwt(
+  public val header: JwtHeader,
+  public val claims: JwtClaimsSet,
+  public val signature: ByteArray,
+  public val parts: List<String>
+) {
+  /**
+   * Verifies the JWT.
+   *
+   */
+  public fun verify() {
+    val decodedJws = DecodedJws(
+      header = header,
+      payload = Convert(parts[1], EncodingFormat.Base64Url).toByteArray(),
+      signature = signature,
+      parts = parts
+    )
+    decodedJws.verify()
+  }
+}
+
+public typealias JwtHeader = JwsHeader
+
+/**
+ * Claims represents JWT (JSON Web Token) Claims
+ * Spec: https://datatracker.ietf.org/doc/html/rfc7519#section-4
+ *
+ * @property iss identifies the principal that issued the
+ * @property sub the principal that is the subject of the JWT.
+ * @property aud the recipients that the JWT is intended for.
+ * @property exp the expiration time on or after which the JWT must not be accepted for processing.
+ * @property nbf the time before which the JWT must not be accepted for processing.
+ * @property iat the time at which the JWT was issued.
+ * @property jti provides a unique identifier for the JWT.
+ * @property misc additional claims (i.e. VerifiableCredential, VerifiablePresentation)
+ */
+@JsonSerialize(using = JwtClaimsSetSerializer::class)
+@JsonDeserialize(using = JwtClaimsSetDeserializer::class)
+public class JwtClaimsSet(
+  public val iss: String? = null,
+  public val sub: String? = null,
+  public val aud: String? = null,
+  public val exp: Long? = null,
```

Comment:
I think long is the right impl here but I think i see other libs with int

Reply:
i agree it should stay as a long - can you tell me which lib has it as an int so we can write an issue for it? i checked js (number), go(int64) and swift (ISO8601Date)